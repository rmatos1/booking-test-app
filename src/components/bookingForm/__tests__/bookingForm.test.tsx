import { describe, test } from "@jest/globals";
import { RenderResult, fireEvent, render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import moment from "moment";

import { BookingForm, IBookingForm } from "..";

const DATE = "2023-12-19";
const GUESTS = 1;

const defaultProps: IBookingForm = {
  onFormSubmit: jest.fn(),
};

const setup = (componentProps?: IBookingForm): JSX.Element => {
  const baseProps = componentProps || defaultProps;

  return (
    <Router>
      <BookingForm {...baseProps} />
    </Router>
  );
};

describe("<BookingForm />", () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = render(setup());
  });

  test("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render the defined title", () => {
    const TITLE = "Booking";

    wrapper = render(setup({ ...defaultProps, title: TITLE }));

    const title = wrapper.getByText(TITLE);

    expect(title).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test("should render the specified date on the check-in input", () => {
    const inputCheckIn = wrapper.getByTestId(
      "input-check-in"
    ) as HTMLInputElement;
    fireEvent.change(inputCheckIn, { target: { value: DATE } });

    expect(inputCheckIn.value).toBe(DATE);
    expect(wrapper).toMatchSnapshot();
  });

  test("should render the specified date on the check-out input", () => {
    const inputCheckOut = wrapper.getByTestId(
      "input-check-out"
    ) as HTMLInputElement;
    fireEvent.change(inputCheckOut, { target: { value: DATE } });

    expect(inputCheckOut.value).toBe(DATE);
    expect(wrapper).toMatchSnapshot();
  });

  test("should render the specified value on the number of guests input", () => {
    const inputQtyGuests = wrapper.getByTestId(
      "input-qty-guests"
    ) as HTMLInputElement;
    fireEvent.change(inputQtyGuests, { target: { value: GUESTS } });

    expect(parseInt(inputQtyGuests.value)).toBe(GUESTS);
    expect(wrapper).toMatchSnapshot();
  });

  test("should display error message when check-in date is before the minimum date", () => {
    const inputCheckIn = wrapper.getByTestId(
      "input-check-in"
    ) as HTMLInputElement;
    const invalidDate = moment().subtract(1, "days").format("YYYY-MM-DD");

    fireEvent.change(inputCheckIn, { target: { value: invalidDate } });
    fireEvent.blur(inputCheckIn);

    const errorMessage = wrapper.getByText("Invalid check-in date");
    expect(errorMessage).toBeDefined();
  });

  test("should display error message when check-out date is before check-in date", () => {
    const inputCheckIn = wrapper.getByTestId(
      "input-check-in"
    ) as HTMLInputElement;
    const inputCheckOut = wrapper.getByTestId(
      "input-check-out"
    ) as HTMLInputElement;

    const checkInDate = moment().add(1, "days").format("YYYY-MM-DD");
    const invalidCheckOutDate = moment(checkInDate)
      .subtract(1, "days")
      .format("YYYY-MM-DD");

    fireEvent.change(inputCheckIn, { target: { value: checkInDate } });
    fireEvent.change(inputCheckOut, { target: { value: invalidCheckOutDate } });
    fireEvent.blur(inputCheckOut);

    const errorMessage = wrapper.getByText("Invalid check-out date");
    expect(errorMessage).toBeDefined();
  });
});
