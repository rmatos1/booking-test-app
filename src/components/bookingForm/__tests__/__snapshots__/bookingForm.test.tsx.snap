// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookingForm /> should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-crrtmM hShWtl"
        data-testid="booking-form"
      >
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Check-in
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-check-in"
            max="Invalid date"
            min="2025-03-01"
            name="checkIn"
            type="date"
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Check-out
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-check-out"
            max="2026-02-28"
            min="2025-03-02"
            name="checkOut"
            type="date"
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Number of guests
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-qty-guests"
            max="4"
            min="1"
            name="guests"
            type="number"
          />
        </div>
        <button
          class="sc-jrAGKZ cntfQH"
          data-testid="check-availability"
        >
          Check Availability
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-crrtmM hShWtl"
      data-testid="booking-form"
    >
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Check-in
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-check-in"
          max="Invalid date"
          min="2025-03-01"
          name="checkIn"
          type="date"
        />
      </div>
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Check-out
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-check-out"
          max="2026-02-28"
          min="2025-03-02"
          name="checkOut"
          type="date"
        />
      </div>
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Number of guests
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-qty-guests"
          max="4"
          min="1"
          name="guests"
          type="number"
        />
      </div>
      <button
        class="sc-jrAGKZ cntfQH"
        data-testid="check-availability"
      >
        Check Availability
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<BookingForm /> should render the defined title 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-crrtmM hShWtl"
        data-testid="booking-form"
      >
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Check-in
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-check-in"
            max="Invalid date"
            min="2025-03-01"
            name="checkIn"
            type="date"
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Check-out
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-check-out"
            max="2026-02-28"
            min="2025-03-02"
            name="checkOut"
            type="date"
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Number of guests
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-qty-guests"
            max="4"
            min="1"
            name="guests"
            type="number"
          />
        </div>
        <button
          class="sc-jrAGKZ cntfQH"
          data-testid="check-availability"
        >
          Check Availability
        </button>
      </form>
    </div>
    <div>
      <form
        class="sc-crrtmM hShWtl"
        data-testid="booking-form"
      >
        <h3
          class="sc-dQpIV fPWcU"
        >
          Booking
        </h3>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Check-in
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-check-in"
            max="Invalid date"
            min="2025-03-01"
            name="checkIn"
            type="date"
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Check-out
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-check-out"
            max="2026-02-28"
            min="2025-03-02"
            name="checkOut"
            type="date"
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Number of guests
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-qty-guests"
            max="4"
            min="1"
            name="guests"
            type="number"
          />
        </div>
        <button
          class="sc-jrAGKZ cntfQH"
          data-testid="check-availability"
        >
          Check Availability
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-crrtmM hShWtl"
      data-testid="booking-form"
    >
      <h3
        class="sc-dQpIV fPWcU"
      >
        Booking
      </h3>
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Check-in
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-check-in"
          max="Invalid date"
          min="2025-03-01"
          name="checkIn"
          type="date"
        />
      </div>
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Check-out
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-check-out"
          max="2026-02-28"
          min="2025-03-02"
          name="checkOut"
          type="date"
        />
      </div>
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Number of guests
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-qty-guests"
          max="4"
          min="1"
          name="guests"
          type="number"
        />
      </div>
      <button
        class="sc-jrAGKZ cntfQH"
        data-testid="check-availability"
      >
        Check Availability
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<BookingForm /> should render the specified date on the check-in input 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-crrtmM hShWtl"
        data-testid="booking-form"
      >
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Check-in
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-check-in"
            max="Invalid date"
            min="2025-03-01"
            name="checkIn"
            type="date"
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Check-out
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-check-out"
            max="2026-02-28"
            min="2023-12-20"
            name="checkOut"
            type="date"
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Number of guests
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-qty-guests"
            max="4"
            min="1"
            name="guests"
            type="number"
          />
        </div>
        <button
          class="sc-jrAGKZ cntfQH"
          data-testid="check-availability"
        >
          Check Availability
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-crrtmM hShWtl"
      data-testid="booking-form"
    >
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Check-in
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-check-in"
          max="Invalid date"
          min="2025-03-01"
          name="checkIn"
          type="date"
        />
      </div>
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Check-out
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-check-out"
          max="2026-02-28"
          min="2023-12-20"
          name="checkOut"
          type="date"
        />
      </div>
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Number of guests
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-qty-guests"
          max="4"
          min="1"
          name="guests"
          type="number"
        />
      </div>
      <button
        class="sc-jrAGKZ cntfQH"
        data-testid="check-availability"
      >
        Check Availability
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<BookingForm /> should render the specified date on the check-out input 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-crrtmM hShWtl"
        data-testid="booking-form"
      >
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Check-in
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-check-in"
            max="2023-12-18"
            min="2025-03-01"
            name="checkIn"
            type="date"
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Check-out
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-check-out"
            max="2026-02-28"
            min="2025-03-02"
            name="checkOut"
            type="date"
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Number of guests
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-qty-guests"
            max="4"
            min="1"
            name="guests"
            type="number"
          />
        </div>
        <button
          class="sc-jrAGKZ cntfQH"
          data-testid="check-availability"
        >
          Check Availability
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-crrtmM hShWtl"
      data-testid="booking-form"
    >
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Check-in
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-check-in"
          max="2023-12-18"
          min="2025-03-01"
          name="checkIn"
          type="date"
        />
      </div>
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Check-out
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-check-out"
          max="2026-02-28"
          min="2025-03-02"
          name="checkOut"
          type="date"
        />
      </div>
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Number of guests
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-qty-guests"
          max="4"
          min="1"
          name="guests"
          type="number"
        />
      </div>
      <button
        class="sc-jrAGKZ cntfQH"
        data-testid="check-availability"
      >
        Check Availability
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<BookingForm /> should render the specified value on the number of guests input 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-crrtmM hShWtl"
        data-testid="booking-form"
      >
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Check-in
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-check-in"
            max="Invalid date"
            min="2025-03-01"
            name="checkIn"
            type="date"
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Check-out
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-check-out"
            max="2026-02-28"
            min="2025-03-02"
            name="checkOut"
            type="date"
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Number of guests
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-qty-guests"
            max="4"
            min="1"
            name="guests"
            type="number"
          />
        </div>
        <button
          class="sc-jrAGKZ cntfQH"
          data-testid="check-availability"
        >
          Check Availability
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-crrtmM hShWtl"
      data-testid="booking-form"
    >
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Check-in
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-check-in"
          max="Invalid date"
          min="2025-03-01"
          name="checkIn"
          type="date"
        />
      </div>
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Check-out
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-check-out"
          max="2026-02-28"
          min="2025-03-02"
          name="checkOut"
          type="date"
        />
      </div>
      <div
        class="sc-jSguLX klDGSv"
      >
        <label
          class="sc-gKsecS dRZBKP"
          data-testid="label"
        >
          Number of guests
        </label>
        <input
          class="sc-iBPTVF YEbKX"
          data-testid="input-qty-guests"
          max="4"
          min="1"
          name="guests"
          type="number"
        />
      </div>
      <button
        class="sc-jrAGKZ cntfQH"
        data-testid="check-availability"
      >
        Check Availability
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
